src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdStateUpdateService.java

```diff
@@ -0,0 +1,118 @@
+            throws UpdateCaseException {
+
+        if (!isHearingCancelled(hearingResponse)) {
+            return;
+        }
+
+        String cancellationReason = hearingResponse.getHearingResponse().getHearingCancellationReason();
+        CancellationReason reason = CancellationReason.getCancellationReasonByLabel(cancellationReason);
+
+        if (isNull(reason)) {
+            throw new UpdateCaseException(String.format("Can not map cancellation reason %s for caseId %s",
+                    cancellationReason, sscsCaseData.getCcdCaseId()));
+        }
+
+        State state = mapHmcCancelledToCcdState(reason, sscsCaseData.getCcdCaseId());
+        setState(sscsCaseData, state);
+    }
+
+    public void updateFailed(@Valid SscsCaseData sscsCaseData) {
+        setState(sscsCaseData, State.HANDLING_ERROR);
+    }
+
+    private void setState(SscsCaseData sscsCaseData, State state) {
+        sscsCaseData.setState(state);
+        log.info("CCD state has been updated to {} for caseId {}", state, sscsCaseData.getCcdCaseId());
+    }
+
+    private State mapHmcCreatedOrUpdatedToCcd(ListingCaseStatus listingCaseStatus, String caseId)
+            throws UpdateCaseException {
+        switch (listingCaseStatus) {
+            case LISTED:
+                return HEARING;
+            case AWAITING_LISTING:
+                return READY_TO_LIST;
+            default:
+                throw new UpdateCaseException(String.format("Can not map HMC updated or create listing status %s "
+                                + "with CCD state for caseId %s",
+                        listingCaseStatus, caseId));
+        }
+    }
+
+    @SuppressWarnings({"PMD.CyclomaticComplexity"})
+    private State mapHmcCancelledToCcdState(CancellationReason cancellationReason, String caseId)
+            throws UpdateCaseException {
+        switch (cancellationReason) {
+            case WITHDRAWN:
+            case STRUCK_OUT:
+            case LAPSED:
+                return DORMANT_APPEAL_STATE;
+            case PARTY_UNABLE_TO_ATTEND:
+            case EXCLUSION:
+            case INCOMPLETE_TRIBUNAL:
+            case LISTED_IN_ERROR:
+            case OTHER:
+            case PARTY_DID_NOT_ATTEND:
+                return READY_TO_LIST;
+            default:
+                throw new UpdateCaseException(String.format("Can not map cancellation reason %s for caseId %s",
+                        cancellationReason, caseId));
+        }
+
+    }
+
+    private static boolean isHearingListingStatusFixed(HearingGetResponse hearingResponse) {
+        return FIXED == hearingResponse.getHearingResponse().getListingStatus();
+    }
+
+    private static boolean isHearingCancelled(HearingGetResponse hearingResponse) {
+        return CANCELLED.equalsIgnoreCase(hearingResponse.getRequestDetails().getStatus())
+            || nonNull(hearingResponse.getHearingResponse().getHearingCancellationReason());
```

Comment:
Minor: Could hearingResponse.getRequestDetails().getStatus() also return an enum type, as in the code immediately preceding? hearingResponse.getHearingResponse().getListingStatus()

Reply:
I don't think there is a enum type yet for getRequestDetails().getStatus, we don't even set it, so I don't know the logic here (I didn't write this)